motr/client_init.c

```diff
@@ -1687,7 +1693,15 @@ int m0_client_init(struct m0_client **m0c_p,
 			size = conf->mc_addb_size;
 			M0_LOG(M0_DEBUG, "ADDB size = %" PRIu64 "", size);
 		}
-		sprintf(buf, "linuxstob:./addb_%d", (int)m0_pid());
+#ifndef __KERNEL__
+		addb_stob_location = getenv("M0_CLIENT_ADDB_DIR");
+#endif
+		if(addb_stob_location == NULL)
+		{
+			addb_stob_location = ".";
+		}
+		sprintf(buf, "linuxstob:%s/addb_%d",
```

Comment:
s/sprintf/snprintf/

Reply:
also assert and log a message is the value of M0_CLIENT_ADDB_DIR is greater than buffer size.